avro/src/ser_direct.rs

```diff
@@ -0,0 +1,2032 @@
+use std::{borrow::Cow, io::Write};
+
+use serde::ser;
+
+use crate::{
+    encode::{encode_int, encode_long},
+    error::Error,
+    schema::{Name, NamesRef, Namespace, RecordSchema, Schema},
+};
+
+const RECORD_FIELD_INIT_BUFFER_SIZE: usize = 64;
+const COLLECTION_SERIALIZER_ITEM_LIMIT: usize = 1024;
+const COLLECTION_SERIALIZER_DEFAULT_INIT_ITEM_CAPACITY: usize = 32;
+const SINGLE_VALUE_INIT_BUFFER_SIZE: usize = 128;
+
+pub struct DirectSerializeSeq<'a, 's, W: Write> {
```

Comment:
Rustdoc please for all public items!
With as much as possible details because even the current ser.rs and de.rs are hard to maintain.